packages/common/src/services/dictionaryService.ts

```diff
@@ -0,0 +1,30 @@
+import { isEmpty } from 'lodash-es';
+import { Dependencies } from '../config/config.js';
+
+import getCategoryUtils from '../utils/categoryUtils.js';
+import getDictionaryUtils from '../utils/dictionaryUtils.js';
+
+const dictionaryService = (dependencies: Dependencies) => {
+	return {
+		register: async (categoryName: string, dictionaryName: string, version: string) => {
```

Comment:
So the return type here is `Dictionary | undefined`. Is the dev supposed to interpret `undefined` as a failure to register the dictionary? This appears to be possible because of 2 reasons:
1. unable to get or create the category
2. unable to fetch the dictionary

These are two different reasons for failure that are not being communicated back to the user, and maybe that is acceptable but I am not sure it is.

A bigger issue here is that the two async operations that htis function relies on could both throw an error (as you coded them) but this function does not have a try/catch block to handle this gracefully. This is my largest issue with using thrown errors to communicate issues in TS: there is no check within the code to ensure you catch all thrown errors and handle them completely. Since you are currently throwing errors though we need to be sure to catch them and handle them here. (Make sure to handle each error type from each function separately ;) )

Reply:
Changed to return only a `Promise<Dictionary>` in case of success. The `undefined` return was removed and now it's handled by a downstream errors. throwing a `ServiceUnavailable` error type in case of failure in downstream services.